### PR TITLE
Split ownership of a8n code into web and core

### PR DESCRIPTION
**Pull request by felixfbecker**
As proposed on Slack. Reduces notification noise.

For changes to behaviour, we can still request review from the whole automation team.

Also made it so that GraphQL API schema changes need to be reviewed by both backend and consumers.

### CODE REVIEW

**Comment by tsenart** [comment]
> If you don't mind: can you also add me to the @sourcegraph/automation-web team? I'd love to be notified of the web PRs.

Side note: Using appropriate PR labels could be a solution too, as it was suggested to allow those wanting to continue to subscribe to all core PRs if the proposal to not ask code review from the entire team moves forward.

File: .github/CODEOWNERS [hunk 1]
```diff
@@ -134,11 +134,10 @@ tslint.config.js @sourcegraph/web
...
 # Automation
-*/a8n/* @sourcegraph/automation
-/enterprise/internal/a8n @sourcegraph/automation
-/internal/a8n @sourcegraph/automation
-**/campaigns/** @sourcegraph/automation
-web/**/campaigns/** @sourcegraph/automation @sourcegraph/web
+/cmd/frontend/graphqlbackend/a8n.go @sourcegraph/automation-core
+/enterprise/internal/a8n @sourcegraph/automation-core
+/internal/a8n @sourcegraph/automation-core
+/web/**/campaigns/** @sourcegraph/automation-web @sourcegraph/web
```
Comment (tsenart) [inline .github/CODEOWNERS]: Are there objections to just notify the @sourcegraph/core-services team for all automation backend code?

**Comment by felixfbecker** [comment]
@mrnugget added you to the list explicitly so that you can review without dismissing the review request from the web team :)

> Side note: Using appropriate PR labels could be a solution too, as it was suggested to allow those wanting to continue to subscribe to all core PRs if the proposal to not ask code review from the entire team moves forward.

I'm not sure I follow. Labels don't trigger any notifications?

**Comment by tsenart** [comment]
> I'm not sure I follow. Labels don't trigger any notifications?

No, but they can be used to filter them, I think.

**Comment by mrnugget** [comment]
> @mrnugget added you to the list explicitly so that you can review without dismissing the review request from the web team :)

Thanks! Just for the record: I don't necessarily want to review, I just want to be notified (which is also why labels don't work for this)

**Comment by felixfbecker** [comment]
@tsenart Yeah, I'm all for labels (and project boards, and milestones). This is purely to solve _notifications_, and we can use all those other things as well 👍

File: .github/CODEOWNERS (hunk 1 above)
Comment (felixfbecker) [reply .github/CODEOWNERS]: I'm fine with that, feel free to open a follow-up PR if @sourcegraph/core-services wants that :)

**Pull request merged**